machine/v1beta1/types_vsphereprovider.go

```diff
@@ -70,6 +70,13 @@ type VSphereMachineProviderSpec struct {
 	// When using LinkedClone, if no snapshots exist for the source template, falls back to FullClone.
 	// +optional
 	CloneMode CloneMode `json:"cloneMode,omitempty"`
+	// Disks is a list of non OS disks to be attached to the VM.
+	// +openshift:enable:FeatureGate=VSphereMultiDisk
+	// +openshift:validation:FeatureGateAwareMaxItems:featureGate="",maxItems=0
+	// +openshift:validation:FeatureGateAwareMaxItems:featureGate=VSphereMultiDisk,maxItems=15
```

Comment:
Make the whole field featuregated, rather than feature gated limits (though note this isn't actually part of a CRD so these markers are just for show at the moment)

Reply:
Ok, I can do that.   Yep, I had to make all this logic in the webhook of MAPI operator due to provider specs not being proper CRDs.